src/mds/FSMap.cc

```diff
@@ -868,7 +872,7 @@ void FSMap::sanity() const
         ceph_assert(fs->mds_map.failed.count(info.rank) == 0);
         ceph_assert(fs->mds_map.damaged.count(info.rank) == 0);
       } else {
-        ceph_assert(fs->mds_map.allows_standby_replay());
+        ceph_assert(!pending || fs->mds_map.allows_standby_replay());
```

Comment:
If it's pending, then we don't do the assertion. Is that what we're trying to do? So basically
```
if (!pending) {
  ceph_assert(fs->mds_map.allows_standby_replay());
} 
```
So shouldn't it be,
``ceph_assert(pending || fs->mds_map.allows_standby_replay());``

Reply:
You got it backwards, it would be:

````
if (pending) {
  ceph_assert(fs->mds_map.allows_standby_replay());
} 
````

(I think the assert is correct...)